arch/risc-v/src/mpfs/mpfs_entrypoints.h

```diff
@@ -0,0 +1,93 @@
+/****************************************************************************
+ * arch/risc-v/src/mpfs/mpfs_entrypoints.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __ARCH_RISCV_SRC_MPFS_MPFS_ENTRYPOINTS_H
+#define __ARCH_RISCV_SRC_MPFS_MPFS_ENTRYPOINTS_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdint.h>
+#include <stdbool.h>
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+#ifdef __cplusplus
```

Comment:
Just remove EXTERN from the function prototypes (like you did) but restore this to what it was.

Look at e.g. riscv_internal.h or riscv_common_memorymap.h
EXTERN is used for global symbols not prototypes.

Reply:
Though, no public symbols in this header (only prototypes). Maybe it is fine like it is. I know that the "final" boss will have an opinion about this :)

Reply:
I'd assume that if you include this from a c++ code, you'll still need the extern "C" for the prototypes so that it doesn't get c++ name mangling and they link against the nuttx C compiled stuff

Reply:
Yes you absolutely do need the guard.

```
#undef EXTERN
#if defined(__cplusplus)
#define EXTERN extern "C"
extern "C"
{
#else
#define EXTERN extern
#endif
```

That is how the guard is in other headers, as you originally did. However, for function prototypes it is not necessary to explicitly use the EXTERN macro, as they are inside the extern "C" { } block.

However, for global symbols you MUST put EXTERN in front, otherwise it is treated as a _definition_ and that is very likely not what you wanted to do.

So the original header was right, with the exception of explicitly adding the EXTERN macro in front.